[ko] Update Korean localization contribution guide

ianychoi opened the pull request:
Revised and updated content in localization_ko.md
- Reflected recent changes and best practices for the Korean localization team

Closes: #52313

<!--
 Hello!

 PLEASE title the FIRST commit appropriately, so that if you squash all
 your commits into one, the combined commit message makes sense.
 For overall help on editing and submitting pull requests, visit:
  https://kubernetes.io/docs/contribute/suggesting-improvements/

 Use the default base branch, “main”, if you're documenting existing
 features in the English localization.

 If you're working on a different localization (not English), see
 https://kubernetes.io/docs/contribute/new-content/overview/#choose-which-git-branch-to-use
 for advice.

 If you're documenting a feature that will be part of a future release, see
 https://kubernetes.io/docs/contribute/new-content/new-features/ for advice.
-->
### Description

<!--
 Remember to ADD A DESCRIPTION and delete this note before submitting
 your pull request. The description should explain what will change,
 and why.
-->

### Issue

<!--
 If this pull request resolves an open issue, please link the issue in the PR
 description so it will automatically close when the PR is merged.

 See the GitHub documentation for more details and other options:

 https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

Closes: #

wonyongg reviewed (commented):
안녕하세요 영락님.

> 1. 리뷰어 및 승인자는 작업 중인 한글화 작업 브랜치(예: dev-1.17-ko.3)에 영향을 최소화 하기 위해서,
신규 한글화 작업 브랜치(예: dev-1.17-ko.4) 생성 시점에 맞춰 확정된 PR을 승인한다.
개선된 한글화 용어집은 신규 한글화 작업 브랜치부터 적용한다.

한글화 용어집 개선 가이드 中
235-237Line 역시 과거 방식인 한글화 작업 브랜치에 대한 내용이 포함되어 있기 때문에
삭제하거나 참고 사항으로 남겨두는 방식이 필요하지 않을까요?

이에 해당 부분을
> 한글화팀 회의를 통해 제안이 확정되면, 리뷰어 및 승인자는 해당 PR을 승인한다.

로 바꾸면 어떨까 제안드립니다.

seokho-son reviewed (commented):
@ianychoi @jongwooo 

개선 제안 감사합니다. 리뷰가 너무 늦어서 죄송합니다.

구조와 내용의 명확성을 개선하여, 다시 제안 드려봅니다.

ianychoi commented:
Thank you @seokho-son - I fully agree with your comments. Also I have added you as co-author. I have squashed into one comment.

@jongwooo @seokho-son : PTAL

seokho-son commented:
/lgtm

seokho-son commented:
/approve

Thanks for this improvement!
